apps/examples/hello/hello_main.c

```diff
@@ -56,17 +56,38 @@
 
 #include <tinyara/config.h>
 #include <stdio.h>
+#include <tinyara/serial/serial.h>
 
 /****************************************************************************
  * hello_main
  ****************************************************************************/
 
+extern struct log_net_chunk_s *log_net_head;
+
+int read_counter = 0;
+
+void read_log_net() {
+	struct log_net_chunk_s *next_chunk = log_net_head;
+	while (next_chunk != NULL) {
+		for (int i = 0; i < next_chunk->size; i++) { /* printing the data to verify storing of logs, later to be replaced by network code to send to server*/
+			lldbg_noarg("%c", next_chunk->arr[i]);
+		}
+		struct log_net_chunk_s *tmp = next_chunk;
+		next_chunk = next_chunk->next;
+		free(tmp); /* once we read the stored data, free the node */
+	}
+	log_net_head = NULL;
+}
+
 #ifdef CONFIG_BUILD_KERNEL
 int main(int argc, FAR char *argv[])
 #else
 int hello_main(int argc, char *argv[])
 #endif
 {
-	printf("Hello, World!!\n");
+	lldbg("Read output %dth time!! ---> starts here <---\n", read_counter);
```

Comment:
I think `hello` is a simple application,
so I think it is better to make a new app for log.